### PR TITLE
Introduce gax package genrator for nodejs.

### PR DESCRIPTION
**Pull request by jmuk**

### CODE REVIEW

**Comment by geigerj** [comment]
Can you update the tests to cover the new templates?

File: templates/gax/nodejs/PUBLISHING.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+== PREREQUISITES
+
+- Node must installed
+- The latest version of [Bundler][] must be installed
+
+
+== TO PUBLISH
+
+- Make sure you have an account on [rubygems][].
```
Comment (geigerj) [inline templates/gax/nodejs/PUBLISHING.md]: Update Ruby reference here and below

Comment (jmuk) [inline templates/gax/nodejs/PUBLISHING.md]: done

**Comment by jmuk** [comment]
Added the test as well -- I've actually found a bug on the existing code through the test. Test is nice always.

File: lib/packager.js [hunk 2]
```diff
@@ -365,15 +366,30 @@ function makeNodejsPackage(opts, done) {
...
+                    'was created in', opts.top);
+        console.log('To publish it, read', path.join(opts.top, 'PUBLISHING.md'),
+                    'for the next steps');
+      }
+      done(err);
+    });
+  }
+
+  // find lib/*.js files for gax API packages to generate index.js properly.
+  glob.glob('*.js', {cwd: path.join(opts.top, 'lib')}, runTask);
```
Comment (geigerj) [inline lib/packager.js]: It looks like `glob` will access the file system even if `opts.mockApiFilesForTest` is specified. Can we skip this step if it is?

File: test/packager.js [hunk 3]
```diff
@@ -394,6 +394,23 @@ describe('the nodejs package builder', function() {
...
+  it ('should construct a gax package', function(done) {
+    var opts = _.merge({
+      packageInfo: testPackageInfo,
+      top: path.join(top, 'nodejs')
+    }, {templateDir: path.join(__dirname, '..', 'templates', 'gax', 'nodejs')});
+    opts.mockApiFilesForTest = ['foo_api.js', 'bar_api.js'];
+    var expanded = [
+      'nodejs/index.js'
+    ];
+    var compareWithFixture =genFixtureCompareFunc(top, ['gax']);
```
Comment (geigerj) [inline test/packager.js]: missing a space after `=`

File: lib/packager.js (hunk 2 above)
Comment (jmuk) [inline lib/packager.js]: done

File: test/packager.js (hunk 3 above)
Comment (jmuk) [inline test/packager.js]: done

**Comment by geigerj** [comment]
LGTM

**Pull request merged**